apps/demos/Demos/Chat/Customization/Angular/app/app.component.ts

```diff
@@ -0,0 +1,55 @@
+import { NgModule, Component, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
+import { DxChatModule, DxCheckBoxModule, DxSelectBoxModule } from 'devextreme-angular';
+import { User, Message, MessageEnteredEvent } from 'devextreme/ui/chat';
+import { AppService } from './app.service';
+import { Observable } from 'rxjs';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+let modulePrefix = '';
+// @ts-ignore
+if (window && window.config.packageConfigPaths) {
+  modulePrefix = '/app';
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: `.${modulePrefix}/app.component.html`,
+  styleUrls: [`.${modulePrefix}/app.component.css`],
+})
+export class AppComponent {
+  currentUser: User;
+  supportAgent: User;
+  messages$: Observable<Message[]>;
+  dayHeaderFormat = ['dd/MM/yyyy', 'dd.MM.yyyy', 'MMMM dd, yyyy', 'EEEE, MMMM dd'];
```

Comment:
Can we split the data separately, as you do in other approaches?

Reply:
Yeah, it can be moved to a service

Reply:
[a40d776](https://github.com/DevExpress/DevExtreme/pull/28482/commits/a40d77698d64bcb42c0340010e88cdccd0628c50)